velox/functions/sparksql/RegexFunctions.cpp

```diff
@@ -78,8 +227,21 @@ std::shared_ptr<exec::VectorFunction> makeRegexExtract(
     const std::vector<exec::VectorFunctionArg>& inputArgs,
     const core::QueryConfig& config) {
   auto result = makeRe2Extract(name, inputArgs, config, /*emptyNoMatch=*/true);
-  ensureRegexIsCompatible("REGEXP_EXTRACT", inputArgs[1].constantValue);
+  ensureRegexIsConstantAndCompatible(
+      "REGEXP_EXTRACT", inputArgs[1].constantValue);
   return result;
 }
 
+void registerRegexReplace(const std::string& prefix) {
+  registerFunction<RegexReplaceFunction, Varchar, Varchar, Varchar, Varchar>(
+      {prefix + "REGEX_REPLACE"});
+  registerFunction<
+      RegexReplaceFunction,
+      Varchar,
+      Varchar,
+      Varchar,
+      Varchar,
+      int64_t>({prefix, "REGEX_REPLACE"});
```

Comment:
@codyschierbeck There is a bug here which causes this function to be registered with 2 names: prefix and "REGEX_REPLACE". The fuzzer uses empty prefix which causes the function to be registered with empty name. This then makes it confusing that an expression doesn't seem to have regex_replace function when in fact it does.

To fix, replace prefix, with prefix +.

We should also add a check to function registry to not allow registering functions with empty names. CC: @laithsakka @kgpai @bikramSingh91 

Would you help make this change?